agent/agent.go

```diff
@@ -240,22 +240,28 @@ func (r *Agent) setupSink(ctx context.Context, sr recipe.SinkRecipe, stream *str
 		return
 	}
 
-	stream.subscribe(func(records []models.Record) (err error) {
-		err = retryIfNeeded(
+	stream.subscribe(func(records []models.Record) error {
+		err := retryIfNeeded(
 			func() error {
 				return sink.Sink(ctx, records)
 			},
 			r.retryTimes,
 			r.retryInitialInterval,
 			func(e error, d time.Duration) {
-				r.logger.Info("retrying sink", "duration", d, "sink", sr.Name, "error", e.Error())
+				r.logger.Info(
+					fmt.Sprintf("retrying sink in %d", d),
+					"sink", sr.Name,
+					"error", e.Error())
 			},
 		)
+		// error (after exhausted retries) will just be skipped and logged
 		if err != nil {
-			return errors.Wrapf(err, "error running sink \"%s\"", sr.Name)
+			r.logger.Error("error running sink", "sink", sr.Name, "error", err.Error())
```

Comment:
@StewartJingga Shall we make this behavior of skipping record after retry exhaust configurable?

Reply:
sure we could, but default would be to skip. wdyt?

Reply:
Skip is data loss. So I would say that should not be default.

Reply:
it is a data loss, but should we stop the run? 🤔 

also for the configuration, do you prefer it on recipe level or on meteor level?

Reply:
Let's keep meteor level for now.